File: hashmap/valid-anagram/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    var cnt [26]int
    for i := 0; i < len(s); i++ {
        cnt[s[i]-'a']++
        cnt[t[i]-'a']--
    }
    for _, v := range cnt[:] {
        if v != 0 {
            return false
        }
    }
    return true
}

func isAnagramUnicode(s string, t string) bool {
	if utf8.RuneCountInString(s) != utf8.RuneCountInString(t) {
        return false
    }

    m := make(map[rune]int, len(s))

    for _, r := range s {
        m[r]++
    }
    for _, r := range t {
        m[r]--
        if m[r] == 0 {
            delete(m, r)
        }
    }
    return len(m) == 0
}

File: hashmap/valid-anagram/valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) return false;
    const cnt = new Array(26).fill(0);

    for (let i = 0; i < s.length; i++) {
        cnt[s.charCodeAt(i) - 97]++;   // 'a' = 97
        cnt[t.charCodeAt(i) - 97]--;
    }
    return cnt.every(x => x === 0);
};

function isAnagramUnicode(s: string, t: string): boolean{
    const map = new Map<string, number>();

    for (const ch of s) map.set(ch, (map.get(ch) ?? 0) + 1);
    for (const ch of t) map.set(ch, (map.get(ch) ?? 0) - 1);

    for (const v of map.values()) if (v !== 0) return false;
    return true;
}